### PR TITLE
CC-7524: Bump Datagen connector version in cp-quickstart

### PR DESCRIPTION
**Pull request by C0urante**
[Jira](https://confluentinc.atlassian.net/browse/CC-7254)

The most recent release of the Datagen connector, 0.1.7, includes a fix that removes the `AvroConverter` class from its dependencies. This prevents that class from being picked up as a converter plugin by the Connect framework, which as of https://github.com/apache/kafka/pull/7315/ is guaranteed to happen on up-to-date versions of the framework. The Datagen connector excludes several transitive dependencies of the Avro converter, so any the version that's packaged with the connector is used as a converter plugin, things break.

### CODE REVIEW

**Comment by C0urante** [comment]
@ybyzek @sdanduConf could one/both of you take a look at this?

**Comment by C0urante** [comment]
Verified locally on CP 5.3.1, 5.3.2-SNAPSHOT, and 5.4.0-SNAPSHOT (using the quickstart on the 5.4.x branch).

**Review by sdandu-gh — APPROVED** [review]
LGTM, thanks!

**Comment by C0urante** [comment]
@ybyzek would you mind taking a look? Don't have merge permission on this repo.

**Comment by ybyzek** [comment]
@C0urante I made this change in `5.4.x` branch via https://github.com/confluentinc/examples/pull/363.

Do you think it is necessary to have it backported in `5.3.x`?

**Comment by C0urante** [comment]
As long as the LOE is as low as just merging this PR, I'd say it's probably best that we do backport to 5.3.x. In fact, it may even be worth it to backport to 5.3.0-post (or further) so that all the quickstarts use the same version of the connector, but that's your call.

**Comment by ybyzek** [comment]
If this change is going to be made to earlier branches, the changes:

1. Must be validated on each of those branches (which you did for cp-quickstart)
2. Should be applied to all instances where it is used, for which there are six demos:

```
ccloud/beginner-cloud/Dockerfile:RUN confluent-hub install --no-prompt confluentinc/kafka-connect-datagen:0.1.6
ccloud/start.sh:confluent-hub install --no-prompt confluentinc/kafka-connect-datagen:0.1.6
cp-quickstart/start.sh:confluent-hub install --no-prompt confluentinc/kafka-connect-datagen:0.1.6
security/rbac/scripts/enable-rbac-connect.sh:echo "confluent-hub install --no-prompt confluentinc/kafka-connect-datagen:0.1.6"
security/rbac/scripts/enable-rbac-connect.sh:confluent-hub install --no-prompt confluentinc/kafka-connect-datagen:0.1.6
security/acls/Dockerfile:RUN confluent-hub install --no-prompt confluentinc/kafka-connect-datagen:0.1.6
```

So I propose we don't backport (and close this PR) given that:

1. This explodes the matrix of testing
2. I already updated to 0.1.7 to all demos in https://github.com/confluentinc/examples/pull/363
3. 5.4.1-post will soon be the default branch

**Pull request closed without merging**
